base/src/main/java/proguard/optimize/inline/BaseLambdaInliner.java

```diff
@@ -0,0 +1,357 @@
+import proguard.classfile.attribute.visitor.AllAttributeVisitor;
+import proguard.classfile.attribute.visitor.AttributeVisitor;
+import proguard.classfile.constant.FieldrefConstant;
+import proguard.classfile.constant.InterfaceMethodrefConstant;
+import proguard.classfile.constant.visitor.ConstantVisitor;
+import proguard.classfile.editor.AccessFixer;
+import proguard.classfile.editor.ClassEditor;
+import proguard.classfile.editor.CodeAttributeEditor;
+import proguard.classfile.editor.ConstantPoolEditor;
+import proguard.classfile.editor.MethodCopier;
+import proguard.classfile.editor.PeepholeEditor;
+import proguard.classfile.instruction.ConstantInstruction;
+import proguard.classfile.instruction.Instruction;
+import proguard.classfile.instruction.InstructionFactory;
+import proguard.classfile.instruction.VariableInstruction;
+import proguard.classfile.instruction.visitor.AllInstructionVisitor;
+import proguard.classfile.instruction.visitor.InstructionCounter;
+import proguard.classfile.instruction.visitor.InstructionOpCodeFilter;
+import proguard.classfile.instruction.visitor.InstructionVisitor;
+import proguard.classfile.util.ClassReferenceInitializer;
+import proguard.classfile.util.ClassUtil;
+import proguard.classfile.util.InternalTypeEnumeration;
+import proguard.classfile.visitor.MemberVisitor;
+import proguard.evaluation.PartialEvaluator;
+import proguard.evaluation.TracedStack;
+import proguard.optimize.inline.lambdalocator.Lambda;
+import proguard.optimize.peephole.MethodInliner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class that given a method that consumes a lambda is able to create a new method which has this lambda inlined. The
+ * caller will still need to replace the call instruction and handle removing arguments provided to the original call
+ * instruction if this class manages to inline the lambda.
+ */
+public abstract class BaseLambdaInliner implements MemberVisitor, InstructionVisitor, ConstantVisitor {
+    private final Clazz consumingClass;
+    private final Method consumingMethod;
+    private final AppView appView;
+    private final Lambda lambda;
+    private final CodeAttributeEditor codeAttributeEditor;
+    private final boolean isStatic;
+    private final int calledLambdaIndex;
+    private final int sizeAdjustedLambdaIndex;
+    private final PartialEvaluator partialEvaluator;
+    private Clazz interfaceClass;
+    private Clazz lambdaClass;
+    private Method lambdaInvokeMethod;
+    private String bridgeDescriptor;
+    private Method inlinedLambdaMethod;
+    private Method staticInvokeMethod;
+    private InterfaceMethodrefConstant referencedInterfaceConstant;
+    private final List<Integer> invokeMethodCallOffsets;
+
+    public BaseLambdaInliner(AppView appView, Clazz consumingClass, Method consumingMethod, int calledLambdaIndex, Lambda lambda) {
+        this.consumingClass = consumingClass;
+        this.consumingMethod = consumingMethod;
+        this.appView = appView;
```

Comment:
Can you pass in only the `programClassPool` and `libraryClassPool` if that's only what you need.